Tests/test_file_png.py

```diff
@@ -291,30 +291,32 @@ def test_save_p_transparent_black(self):
         self.assert_image(im, "RGBA", (10, 10))
         self.assertEqual(im.getcolors(), [(100, (0, 0, 0, 0))])
 
-    def test_save_l_transparency(self):
-        # There are 559 transparent pixels in l_trns.png.
-        num_transparent = 559
-
-        in_file = "Tests/images/l_trns.png"
-        im = Image.open(in_file)
-        self.assertEqual(im.mode, "L")
-        self.assertEqual(im.info["transparency"], 255)
+    def test_save_greyscale_transparency(self):
+        for mode, num_transparent in {
+            "1": 1994,
+            "L": 559,
+            "I": 559
```

Comment:
```suggestion
            "I": 559,
```

Reply:
I included your change, but managed to lose the specific commit during a rebase, so the change is combined into the one commit now.